salt/client/ssh/__init__.py

```diff
@@ -782,6 +783,10 @@ def __init__(
         self.serial = salt.payload.Serial(opts)
         self.wfuncs = salt.loader.ssh_wrapper(opts, None, self.context)
         self.shell = salt.client.ssh.shell.gen_shell(opts, **args)
+        if self.winrm:
+            # Determine if Windows client is x86 or AMD64
+            arch, _, _ = self.shell.exec_cmd('powershell $ENV:PROCESSOR_ARCHITECTURE')
```

Comment:
Power shell is a very heavy way of doing this.

Here are a few alternatives
```
import platform
platform.machine()
```

What about environment variables.
PROCESSOR_ARCHITECTURE=AMD64
PROCESSOR_IDENTIFIER=Intel64 Family 6 Model 42 Stepping 7, GenuineIntel
PROCESSOR_LEVEL=6
PROCESSOR_REVISION=2a07
ProgramData=C:\ProgramData
ProgramFiles=C:\Program Files
ProgramFiles(x86)=C:\Program Files (x86)
ProgramW6432=C:\Program Files

https://ss64.com/nt/syntax-64bit.html

https://msdn.microsoft.com/en-us/library/aa384274.aspx

Power shell is not a requirement for Salt?

From https://stackoverflow.com/questions/12322308/batch-file-to-check-64bit-or-32bit-os
> PROCESSOR_ARCHITECTURE=x86
> Will appear on Win32, and
> PROCESSOR_ARCHITECTURE=AMD64
> will appear for Win64.
> If you are perversely running the 32-bit cmd.exe process then Windows presents two environment variables:
> PROCESSOR_ARCHITECTURE=x86
> PROCESSOR_ARCHITEW6432=AMD64

Reply:
@damon-atkins, thanks for the idea, but in this instance we're running a remote command on a Windows machine that likely doesn't have a python interpreter yet.